crypto/engine/eng_lib.c

```diff
@@ -91,6 +92,7 @@ int engine_free_util(ENGINE *e, int not_locked)
     if (e->destroy)
         e->destroy(e);
     CRYPTO_free_ex_data(CRYPTO_EX_INDEX_ENGINE, e, &e->ex_data);
+    engine_remove_dynamic_id(e, not_locked);
```

Comment:
hmm, in the debugger I see, that the DSO is unloaded in the CRYPTO_free_ex_data above,
so the dynamic_id should better be unlinked before that.
Thereforfe, I'm going to swap those two statements.